src/modules/fancyzones/FancyZonesLib/util.cpp

```diff
@@ -300,6 +300,38 @@ namespace FancyZonesUtils
         }
     }
 
+    void SizeWindowToRect(HWND window, RECT rect, HWND windowOfRect) noexcept
+    {
+        RECT newWindowRect = rect;
+
+        RECT windowRect{};
+        ::GetWindowRect(window, &windowRect);
+
+        // Take care of borders
+        RECT frameRect{};
+        if (SUCCEEDED(DwmGetWindowAttribute(window, DWMWA_EXTENDED_FRAME_BOUNDS, &frameRect, sizeof(frameRect))))
+        {
+            LONG leftMargin = frameRect.left - windowRect.left;
+            LONG rightMargin = frameRect.right - windowRect.right;
+            LONG bottomMargin = frameRect.bottom - windowRect.bottom;
+            newWindowRect.left -= leftMargin;
+            newWindowRect.right -= rightMargin;
+            newWindowRect.bottom -= bottomMargin;
+        }
+
+        // Take care of windows that cannot be resized
+        if ((::GetWindowLong(window, GWL_STYLE) & WS_SIZEBOX) == 0)
+        {
+            newWindowRect.right = newWindowRect.left + (windowRect.right - windowRect.left);
+            newWindowRect.bottom = newWindowRect.top + (windowRect.bottom - windowRect.top);
+        }
+
+        // Convert to screen coordinates
+        MapWindowRect(windowOfRect, nullptr, &newWindowRect);
+
+        SizeWindowToRect(window, newWindowRect);
```

Comment:
It would be cleaner if you return the `newWindowRect` and call `SizeWindowToRect` separately.

Reply:
In that case, the function would only be usable if called right before `SizeWindowToRect(HWND,RECT)`. That would be OK, but I really do not have a name for the function other than something with "prepare", which immediately makes me think I am doing something wrong. Do you have a name in mind?

The best I found is `AdjustRectForSizeWindowToRect(HWND window, RECT rect, HWND windowOfRect)`.

Reply:
`AdjustRectForSizeWindowToRect` is good IMO.